core/src/main/scala/org/apache/spark/scheduler/DAGScheduler.scala

```diff
@@ -1235,6 +1235,7 @@ class DAGScheduler(
       case FetchFailed(bmAddress, shuffleId, mapId, reduceId, failureMessage) =>
         val failedStage = stageIdToStage(task.stageId)
         val mapStage = shuffleIdToMapStage(shuffleId)
+        var abortedStage = false
```

Comment:
Could you move this line just above `if (disallowStageRetryForTest) {` since it's only used in that scope.